protos/sigstore_common.proto

```diff
@@ -66,9 +66,22 @@ enum PublicKeyDetails {
         PKCS1_RSA_PSS = 2 [deprecated = true]; // See RFC8017
         PKIX_RSA_PKCS1V5 = 3 [deprecated = true];
         PKIX_RSA_PSS = 4 [deprecated = true];
-        PKIX_RSA_PKCS1_2048_SHA256 = 9;
-        PKIX_RSA_PKCS1_3072_SHA256 = 10;
-        PKIX_RSA_PKCS1_4096_SHA256 = 11;
+        // RSA public key in PKIX format, PKCS#1v1.5 signature
+        PKIX_RSA_PKCS1V15_2048_SHA256 = 9;
+        PKIX_RSA_PKCS1V15_3072_SHA256 = 10;
+        PKIX_RSA_PKCS1V15_4096_SHA256 = 11;
+        // RSA public key in PKIX format, RSASSA-PSS signature
+        PKIX_RSA_PSS_2048_SHA256 = 16; // See RFC4055
+        PKIX_RSA_PSS_3072_SHA256 = 17;
+        PKIX_RSA_PSS_4096_SHA256 = 18;
+        // RSA public key in PKCS#1 format, PKCS#1v1.5 signature
+        PKCS1_RSA_PKCS1V15_2048_SHA256 = 19;
+        PKCS1_RSA_PKCS1V15_3072_SHA256 = 20;
+        PKCS1_RSA_PKCS1V15_4096_SHA256 = 21;
+        // RSA public key in PKCS#1 format, RSASSA-PSS signature
+        PKCS1_RSA_PSS_2048_SHA256 = 22; // See RFC4055
+        PKCS1_RSA_PSS_3072_SHA256 = 23;
+        PKCS1_RSA_PSS_4096_SHA256 = 24;
```

Comment:
Here's the full linearization of supported RSA key format + signature format options.

As a random thought: given that the PKIX encoding is just an encapsulation of the PKCS1 encoding for RSA public keys, maybe it makes sense to encourage the ecosystem as a whole to uniformly use the PKIX encoding? In other words: rather than explicitly enumerating the `PKCS1_RSA_*` variants, maybe we could say that they're subsumed under the deprecated `PKCS1_RSA_PKCS1V5` variant and that implementations *should* transform their public keys into PKIX instead?

(I don't know if this is a good idea or not, it just came to mind as an option if we want to reduce the enum size here 🙂)

Reply:
For the public instance, we will only use PKIX. PKCS1 would cover private deployments. Do you know how much support there is for PKCS1 encodings in other languages? As in, if the only sigstore client that supports pkcs1 encoded rsa keys is cosign, maybe we drop these for now?

Reply:
>  Do you know how much support there is for PKCS1 encodings in other languages? As in, if the only sigstore client that supports pkcs1 encoded rsa keys is cosign, maybe we drop these for now?

Not sure about JS or Java, but we don't support PKCS1-encoded keys in `sigstore-python`. CC @bdehamer @loosebazooka @kommendorkapten for additional datapoints here 🙂

Reply:
I would to ignore PKCS1 encoded keys (that's why I didn't add them, only kept the old definitions as `deprecated`). I'll look into JS.

Reply:
sigstore-js only supports PKIX keys now: https://github.com/sigstore/sigstore-js/blob/ed19022f2817c8444cc343984609102d0324fe4f/packages/core/src/crypto.ts#L25

Side note is that nodejs documentation also recommends using PKIX for persisting public keys for long-term storage.

That said it would be possible to enable usage of PKCS1 encoded keys in sigstore-js, but as mentioned earlier, I rather not.

Reply:
Makes sense! Given that, I think we can entirely drop the PKCS1 variants here (and re-add later, if needed).